test/gst-va/encode/vp9.py

```diff
@@ -0,0 +1,169 @@
+      codec         = "vp9",
+      gstdecoder    = "vavp9dec",
+      gstmediatype  = "video/x-vp9",
+      gstparser     = "vp9parse",
+      gstmuxer      = "matroskamux",
+      gstdemuxer    = "matroskademux",
+    )
+
+  def get_file_ext(self):
+    return "webm"
+
+@slash.requires(*platform.have_caps("encode", "vp9_8"))
+class VP9EncoderTest(VP9EncoderBaseTest):
+  def before(self):
+    super().before()
+    vars(self).update(
+      caps          = platform.get_caps("encode", "vp9_8"),
+      lowpower      = False,
+      gstencoder    = "vavp9lpenc",
+    )
+
+@slash.requires(*have_gst_element("vavp9lpenc"))
+@slash.requires(*platform.have_caps("vdenc", "vp9_8"))
+class VP9EncoderLPTest(VP9EncoderBaseTest):
+  def before(self):
+    super().before()
+    vars(self).update(
+      caps          = platform.get_caps("vdenc", "vp9_8"),
+      lowpower      = True,
+      gstencoder    = "vavp9lpenc",
+    )
+
+class cqp(VP9EncoderTest):
+  def init(self, tspec, case, ipmode, qp, quality, refmode, looplvl, loopshp):
+    vars(self).update(tspec[case].copy())
+    vars(self).update(
+      case      = case,
+      gop       = 30 if ipmode != 0 else 1,
+      looplvl   = looplvl,
+      loopshp   = loopshp,
+      qp        = qp,
+      quality   = quality,
+      rcmode    = "cqp",
+    )
+
+  @slash.parametrize(*gen_vp9_cqp_parameters(spec))
+  def test(self, case, ipmode, qp, quality, refmode, looplvl, loopshp):
+    self.init(spec, case, ipmode, qp, quality, refmode, looplvl, loopshp)
+    self.encode()
+
+class cbr(VP9EncoderTest):
+  def init(self, tspec, case, gop, bitrate, fps, refmode, looplvl, loopshp):
+    vars(self).update(tspec[case].copy())
+    vars(self).update(
+      bitrate   = bitrate,
+      case      = case,
+      fps       = fps,
+      frames    = vars(self).get("brframes", self.frames),
+      gop       = gop,
+      looplvl   = looplvl,
+      loopshp   = loopshp,
+      maxrate   = bitrate,
+      minrate   = bitrate,
+      rcmode    = "cbr",
+    )
+ 
+  @slash.parametrize(*gen_vp9_cbr_parameters(spec))
+  def test(self, case, gop, bitrate, fps, refmode, looplvl, loopshp):
+    self.init(spec, case, gop, bitrate, fps, refmode, looplvl, loopshp)
+    self.encode()
+
+class vbr(VP9EncoderTest):
+  def init(self, tspec, case, gop, bitrate, fps, refmode, quality, looplvl, loopshp):
+    vars(self).update(tspec[case].copy())
+    vars(self).update(
+      bitrate   = bitrate,
+      case      = case,
+      fps       = fps,
+      frames    = vars(self).get("brframes", self.frames),
+      gop       = gop,
+      looplvl   = looplvl,
+      loopshp   = loopshp,
+      maxrate   = bitrate,
+      minrate   = bitrate,
+      quality   = quality,
+      rcmode    = "vbr",
+    )
+  
+  @slash.parametrize(*gen_vp9_vbr_parameters(spec))
+  def test(self, case, gop, bitrate, fps, refmode, quality, looplvl, loopshp):
+    self.init(spec, case, gop, bitrate, fps, refmode, quality, looplvl, loopshp)
+    self.encode()
```

Comment:
`vavp9enc` does not have `refmode` property... please use `@parametrize_with_unused`, then remove that parameter from the init function.